src/components/Square.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react'
+
+//component
+const Square = ({
+  id,
+  currentTurn,
+  value,
```

Comment:
this name is really generic and conflicts with your stateValue hook, I'd name it something more descriptive (and keep the same name when passing it between components)